src/components/BaseLayerToggler/BaseLayerToggler.js

```diff
@@ -97,19 +107,30 @@ class BaseLayerToggler extends Component {
 
     if (layerVisible && !prevState.layerVisible) {
       this.next();
-    } else if (layerVisible !== prevState.layerVisible) {
+    } else if (
+      layerVisible !== prevState.layerVisible &&
+      layers.includes(prevState.layerVisible)
+    ) {
       // In case the visibility of the background Layer is change from another component.
       this.toggle(prevState.layerVisible);
     }
 
-    if (this.map && idx !== prevState.idx) {
+    if (
+      this.map &&
+      (idx !== prevState.idx ||
+        BaseLayerToggler.isDifferentLayers(prevState.layers, layers))
+    ) {
       this.map.getLayers().clear();
 
       if (!layers.length) {
         return;
       }
-      const children = layers[idx].getChildren();
-      const childLayers = children.length ? children : [layers[idx]];
+      let children = [];
+      let childLayers = [];
+      if (idx !== null) {
+        children = layers[idx].getChildren();
+        childLayers = children.length ? children : [layers[idx]];
```

Comment:
could create a [undefined] array if layers is empty

Reply:
The is a return if !layers.length just above, so it should be safe isn't it?

Reply:
good point :+1:

Reply:
but `layers[idx]` could be null ?

Reply:
I added a condition to prevent this scenario